tools/lzcomp.c

```diff
@@ -43,8 +43,8 @@ struct command * select_command_sequence(struct command **, const unsigned short
 struct command * merge_command_sequences(const struct command *, unsigned short, const struct command *, unsigned short, unsigned short *);
 unsigned short compressed_length(const struct command *, unsigned short);
 
-int main (int argc, char ** argv) {
-  struct options options = get_options(argc, argv);
+int main (int argc __attribute__((unused)), char ** argv) {
```

Comment:
I’m fine with the others, but is this one really necessary?

Reply:
Would you prefer using `-Wno-unused-parameter`?

Reply:
I mean, does an unused argc actually trigger a warning?

Reply:
I mean, why not actually use it by simply `if (argc <= 1) usage(argv[0]);`?

Reply:
That's a redundant check, since the lack of a filename is already tested for by checking for the null pointer that the C standard says must exist at `argv[argc]`.

Reply:
@iimarckus There is a warning, which is why I silenced it with `__attribute__((unused))`. @mid-kid @aaaaaa123456789 True, instead of `const char * program_name = *argv; <argv parsing loop> if (!*argv) usage(program_name);`, you could have `if (argc == 1) usage(argv[0]); <argv parsing loop>`.

Reply:
Not really, since it can have multiple arguments and still lack the one required argument. It was a crappy suggestion as an alternative to `__attribute__((unused))`